apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/transaction/types/smc/SmcPublishContractTransactionType.java

```diff
@@ -0,0 +1,181 @@
+        BigInteger calculatedFuel = pcf.calcFuel(smartContract);
+        if (!smartContract.getFuel().tryToCharge(calculatedFuel)) {
+            log.error("Needed fuel={} but actual={}", calculatedFuel, smartContract.getFuel());
+            throw new AplUnacceptableTransactionValidationException("Not enough fuel to execute this transaction, expected=" + calculatedFuel + " but actual=" + smartContract.getFuel(), transaction);
+        }
+        //syntactical and semantic validation
+        SmcContractTxProcessor processor = new PublishContractTxValidator(smartContract, context);
+        try {
+            processor.process();
+        } catch (PolyglotException e) {
+            log.debug("SMC: doStateIndependentValidation = INVALID txId={}", transaction.getStringId());
+            throw new AplUnacceptableTransactionValidationException(e.getMessage(), transaction);
+        }
+        if (processor.getExecutionLog().hasError()) {
+            log.debug("SMC: doStateIndependentValidation = INVALID txId={}", transaction.getStringId());
+            throw new AplUnacceptableTransactionValidationException(processor.getExecutionLog().toJsonString(), transaction);
+        }
+        log.debug("SMC: doStateIndependentValidation = VALID txId={}", transaction.getStringId());
+    }
+
+    @Override
+    public void applyAttachment(Transaction transaction, Account senderAccount, Account recipientAccount) {
+        log.debug("SMC: applyAttachment: publish smart contract and call constructor, txId={}", transaction.getStringId());
+        SmartContract smartContract = contractToolService.createNewContract(transaction);
+        SmcPublishContractAttachment attachment = (SmcPublishContractAttachment) transaction.getAttachment();
+        var context = smcConfig.asContext(blockchain.getHeight(),
+            smartContract,
+            integratorFactory.createProcessor(transaction, attachment, senderAccount, recipientAccount, getLedgerEvent())
+        );
+        log.debug("Before processing Address={} Fuel={}", smartContract.getAddress(), smartContract.getFuel());
+        var contractService = new SmcPostponedContractServiceImpl(contractRepository);
+
+        executeContract(transaction, senderAccount, smartContract, new PublishContractTxProcessor(smartContract, context));
+
+        //save contract and contract state
+        contractService.saveContract(smartContract);
+        log.info("Contract {} published init=[{}], txId={}, fuel={}, amountATM={}, owner={}",
+            smartContract.getAddress(), smartContract.getInitCode(), Long.toUnsignedString(transaction.getId()),
+            smartContract.getFuel(), transaction.getAmountATM(), smartContract.getOwner());
+        contractService.commitContractChanges(transaction);
+        log.trace("Changes were committed, txId={}", transaction.getStringId());
+    }
+
+    @Override
+    public boolean mustHaveRecipient() {
+        return false;
```

Comment:
It must not have recipient  but parent class define canHaveRecepient as tru

Reply:
overwritten the canHaveRecepient method